airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/DestinationConsumer.java

```diff
@@ -26,18 +26,20 @@
 
 import io.airbyte.commons.functional.CheckedConsumer;
 
-/**
- * Lifecyle:
- * <p>
- * 1. Consumer object is instantiated.
- * </p>
- * <p>
- * 2. It received messages via {@link DestinationConsumer#accept(T)}
- * </p>
- * <p>
- * 3. Always (on success or failure) finalize by calling {@link DestinationConsumer#close()}
- * </p>
- *
- * @param <T> - type of the message to be consumed.
- */
-public interface DestinationConsumer<T> extends CheckedConsumer<T, Exception>, AutoCloseable {}
+// Lifecycle:
+// 1. Instantiate consumer.
+// 2. start() to initialize any resources that need to be created BEFORE the consumer consumes any
+// messages.
+// 2. consumes ALL records via {@link DestinationConsumer#accept(T)}
+// 3. Always (on success or failure) finalize by calling {@link DestinationConsumer#close()}
+public interface DestinationConsumer<T> extends CheckedConsumer<T, Exception>, AutoCloseable {
+
+  void start() throws Exception;
```

Comment:
Adding this allows us to encapsulate all of the resources used in destination#write in a single class more clearly (instead of instantiating a bunch of them before passing them into the DestinationConsumer constructor. I'm liking this pattern better for the destinations I've touched so far. Can follow up and refactor other destinations to use this pattern if we like it. For now the destinations I wasn't focused on just implement this as a no op (e.g. csv, json, bigquery).